In [1]:
import numpy as np
import pandas as pd
pd.options.display.max_columns = None
pd.options.display.max_rows = None
import matplotlib.pyplot as plt
plt.style.use('seaborn')
import warnings
warnings.filterwarnings('ignore')
import seaborn as sns
import math, time, random, datetime
import pickle

In [2]:
import ads
ads.set_auth(auth='resource_principal')

import oci
from oci.data_science import DataScienceClient
rps = oci.auth.signers.get_resource_principals_signer()
dsc = DataScienceClient(config={}, signer=rps)



In [3]:
from ads.dataset.factory import DatasetFactory
from ads.dataset.dataset_browser import DatasetBrowser
import ads

train = pd.read_csv('UNSW_NB15_training-set.csv')
test = pd.read_csv('UNSW_NB15_testing-set.csv')
test.head()

id       dur proto service state  spkts  dpkts  sbytes  dbytes       rate  \
0   1  0.121478   tcp       -   FIN      6      4     258     172  74.087490   
1   2  0.649902   tcp       -   FIN     14     38     734   42014  78.473372   
2   3  1.623129   tcp       -   FIN      8     16     364   13186  14.170161   
3   4  1.681642   tcp     ftp   FIN     12     12     628     770  13.677108   
4   5  0.449454   tcp       -   FIN     10      6     534     268  33.373826   

   sttl  dttl         sload          dload  sloss  dloss      sinpkt  \
0   252   254  14158.942380    8495.365234      0      0   24.295600   
1    62   252   8395.112305  503571.312500      2     17   49.915000   
2    62   252   1572.271851   60929.230470      1      6  231.875571   
3    62   252   2740.178955    3358.622070      1      3  152.876547   
4   254   252   8561.499023    3987.059814      2      1   47.750333   

       dinpkt          sjit          djit  swin       stcpb       dtcpb  dwin  \
0    8.375000     30.177547     11.830604   255   621772692  2202533631   255   
1   15.432865     61.426934   1387.778330   255  1417884146  3077387971   255   
2  102.737203  17179.586860  11420.926230   255  2116150707  2963114973   255   
3   90.235726    259.080172   4991.784669   255  1107119177  1047442890   255   
4   75.659602   2415.837634    115.807000   255  2436137549  1977154190   255   

     tcprtt    synack    ackdat  smean  dmean  trans_depth  response_body_len  \
0  0.000000  0.000000  0.000000     43     43            0                  0   
1  0.000000  0.000000  0.000000     52   1106            0                  0   
2  0.111897  0.061458  0.050439     46    824            0                  0   
3  0.000000  0.000000  0.000000     52     64            0                  0   
4  0.128381  0.071147  0.057234     53     45            0                  0   

   ct_srv_src  ct_state_ttl  ct_dst_ltm  ct_src_dport_ltm  ct_dst_sport_ltm  \
0           1             0           1                 1                 1   
1          43             1           1                 1                 1   
2           7             1           2                 1                 1   
3           1             1           2                 1                 1   
4          43             1           2                 2                 1   

   ct_dst_src_ltm  is_ftp_login  ct_ftp_cmd  ct_flw_http_mthd  ct_src_ltm  \
0               1             0           0                 0           1   
1               2             0           0                 0           1   
2               3             0           0                 0           2   
3               3             1           1                 0           2   
4              40             0           0                 0           2   

   ct_srv_dst  is_sm_ips_ports attack_cat  label  
0           1                0     Normal      0  
1           6                0     Normal      0  
2           6                0     Normal      0  
3           1                0     Normal      0  
4          39                0     Normal      0

In [4]:
train.head()

id       dur proto service state  spkts  dpkts  sbytes  dbytes  \
0   1  0.000011   udp       -   INT      2      0     496       0   
1   2  0.000008   udp       -   INT      2      0    1762       0   
2   3  0.000005   udp       -   INT      2      0    1068       0   
3   4  0.000006   udp       -   INT      2      0     900       0   
4   5  0.000010   udp       -   INT      2      0    2126       0   

          rate  sttl  dttl        sload  dload  sloss  dloss  sinpkt  dinpkt  \
0   90909.0902   254     0  180363632.0    0.0      0      0   0.011     0.0   
1  125000.0003   254     0  881000000.0    0.0      0      0   0.008     0.0   
2  200000.0051   254     0  854400000.0    0.0      0      0   0.005     0.0   
3  166666.6608   254     0  600000000.0    0.0      0      0   0.006     0.0   
4  100000.0025   254     0  850400000.0    0.0      0      0   0.010     0.0   

   sjit  djit  swin  stcpb  dtcpb  dwin  tcprtt  synack  ackdat  smean  dmean  \
0   0.0   0.0     0      0      0     0     0.0     0.0     0.0    248      0   
1   0.0   0.0     0      0      0     0     0.0     0.0     0.0    881      0   
2   0.0   0.0     0      0      0     0     0.0     0.0     0.0    534      0   
3   0.0   0.0     0      0      0     0     0.0     0.0     0.0    450      0   
4   0.0   0.0     0      0      0     0     0.0     0.0     0.0   1063      0   

   trans_depth  response_body_len  ct_srv_src  ct_state_ttl  ct_dst_ltm  \
0            0                  0           2             2           1   
1            0                  0           2             2           1   
2            0                  0           3             2           1   
3            0                  0           3             2           2   
4            0                  0           3             2           2   

   ct_src_dport_ltm  ct_dst_sport_ltm  ct_dst_src_ltm  is_ftp_login  \
0                 1                 1               2             0   
1                 1                 1               2             0   
2                 1                 1               3             0   
3                 2                 1               3             0   
4                 2                 1               3             0   

   ct_ftp_cmd  ct_flw_http_mthd  ct_src_ltm  ct_srv_dst  is_sm_ips_ports  \
0           0                 0           1           2                0   
1           0                 0           1           2                0   
2           0                 0           1           3                0   
3           0                 0           2           3                0   
4           0                 0           2           3                0   

  attack_cat  label  
0     Normal      0  
1     Normal      0  
2     Normal      0  
3     Normal      0  
4     Normal      0

In [5]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 82332 entries, 0 to 82331
Data columns (total 45 columns):
id                   82332 non-null int64
dur                  82332 non-null float64
proto                82332 non-null object
service              82332 non-null object
state                82332 non-null object
spkts                82332 non-null int64
dpkts                82332 non-null int64
sbytes               82332 non-null int64
dbytes               82332 non-null int64
rate                 82332 non-null float64
sttl                 82332 non-null int64
dttl                 82332 non-null int64
sload                82332 non-null float64
dload                82332 non-null float64
sloss                82332 non-null int64
dloss                82332 non-null int64
sinpkt               82332 non-null float64
dinpkt               82332 non-null float64
sjit                 82332 non-null float64
djit                 82332 non-null float64
swin                 82332 non-n

In [6]:
data = pd.concat([train,test]).reset_index(drop=True)
cols_cat = data.select_dtypes(include = ['object']).columns 
cols_numeric = data._get_numeric_data().columns

In [7]:
data.describe()

id            dur          spkts          dpkts  \
count  257673.000000  257673.000000  257673.000000  257673.000000   
mean    72811.823858       1.246715      19.777144      18.514703   
std     48929.917641       5.974305     135.947152     111.985965   
min         1.000000       0.000000       1.000000       0.000000   
25%     32210.000000       0.000008       2.000000       0.000000   
50%     64419.000000       0.004285       4.000000       2.000000   
75%    110923.000000       0.685777      12.000000      10.000000   
max    175341.000000      59.999989   10646.000000   11018.000000   

             sbytes        dbytes          rate           sttl           dttl  \
count  2.576730e+05  2.576730e+05  2.576730e+05  257673.000000  257673.000000   
mean   8.572952e+03  1.438729e+04  9.125391e+04     180.000931      84.754957   
std    1.737739e+05  1.461993e+05  1.603446e+05     102.488268     112.762131   
min    2.400000e+01  0.000000e+00  0.000000e+00       0.000000       0.000000   
25%    1.140000e+02  0.000000e+00  3.078928e+01      62.000000       0.000000   
50%    5.280000e+02  1.780000e+02  2.955665e+03     254.000000      29.000000   
75%    1.362000e+03  1.064000e+03  1.250000e+05     254.000000     252.000000   
max    1.435577e+07  1.465753e+07  1.000000e+06     255.000000     254.000000   

              sload         dload          sloss          dloss  \
count  2.576730e+05  2.576730e+05  257673.000000  257673.000000   
mean   7.060869e+07  6.582143e+05       4.889317       6.743691   
std    1.857313e+08  2.412372e+06      65.574953      53.702222   
min    0.000000e+00  0.000000e+00       0.000000       0.000000   
25%    1.231800e+04  0.000000e+00       0.000000       0.000000   
50%    7.439423e+05  1.747441e+03       0.000000       0.000000   
75%    8.000000e+07  2.210538e+04       3.000000       2.000000   
max    5.988000e+09  2.242273e+07    5319.000000    5507.000000   

              sinpkt         dinpkt          sjit           djit  \
count  257673.000000  257673.000000  2.576730e+05  257673.000000   
mean      912.300834      98.915462  5.419373e+03     582.251456   
std      6922.153239    1094.048691  4.903450e+04    3930.153369   
min         0.000000       0.000000  0.000000e+00       0.000000   
25%         0.008000       0.000000  0.000000e+00       0.000000   
50%         0.381696       0.007000  6.736370e-01       0.000000   
75%        58.094727      56.438859  2.787367e+03     119.712937   
max     84371.496000   57739.240000  1.483831e+06  463199.240100   

                swin         stcpb         dtcpb           dwin  \
count  257673.000000  2.576730e+05  2.576730e+05  257673.000000   
mean      121.753661  1.006120e+09  1.002295e+09     119.254629   
std       127.367443  1.367795e+09  1.363877e+09     127.230477   
min         0.000000  0.000000e+00  0.000000e+00       0.000000   
25%         0.000000  0.000000e+00  0.000000e+00       0.000000   
50%         0.000000  0.000000e+00  0.000000e+00       0.000000   
75%       255.000000  2.007375e+09  1.992752e+09     255.000000   
max       255.000000  4.294959e+09  4.294882e+09     255.000000   

              tcprtt         synack         ackdat          smean  \
count  257673.000000  257673.000000  257673.000000  257673.000000   
mean        0.046038       0.023652       0.022386     137.639027   
std         0.092908       0.053856       0.045771     205.901118   
min         0.000000       0.000000       0.000000      24.000000   
25%         0.000000       0.000000       0.000000      57.000000   
50%         0.000000       0.000000       0.000000      73.000000   
75%         0.082082       0.036842       0.044665     100.000000   
max         3.821465       3.226788       2.928778    1504.000000   

               dmean    trans_depth  response_body_len     ct_srv_src  \
count  257673.000000  257673.000000       2.576730e+05  257673.000000   
mean      121.649703       0.102242       1.968900e+03       9.383176   
std 

 # Missing Values

### Check and replace the missing values

In [8]:
print(data.isnull().sum())

id                   0
dur                  0
proto                0
service              0
state                0
spkts                0
dpkts                0
sbytes               0
dbytes               0
rate                 0
sttl                 0
dttl                 0
sload                0
dload                0
sloss                0
dloss                0
sinpkt               0
dinpkt               0
sjit                 0
djit                 0
swin                 0
stcpb                0
dtcpb                0
dwin                 0
tcprtt               0
synack               0
ackdat               0
smean                0
dmean                0
trans_depth          0
response_body_len    0
ct_srv_src           0
ct_state_ttl         0
ct_dst_ltm           0
ct_src_dport_ltm     0
ct_dst_sport_ltm     0
ct_dst_src_ltm       0
is_ftp_login         0
ct_ftp_cmd           0
ct_flw_http_mthd     0
ct_src_ltm           0
ct_srv_dst           0
is_sm_ips_ports      0
attack_cat 

All the data is clean and no missing values found in the data sets

In [9]:
data.proto.value_counts()

tcp            123041
udp             92701
unas            15599
arp              3846
ospf             3271
sctp             1474
any               396
gre               313
rsvp              264
ipv6              262
sun-nd            255
swipe             253
pim               253
mobile            253
sep               251
leaf-1            133
xns-idp           133
prm               133
idpr              132
leaf-2            132
ipcv              132
etherip           132
dcn               132
ttp               132
pipe              132
sat-mon           132
compaq-peer       132
ptp               132
xtp               132
cpnx              132
sdrp              132
scps              132
bbn-rcc           132
sps               132
ipv6-route        132
isis              132
srp               132
rvd               132
i-nlsp            132
ddp               132
aris              132
vmtp              132
micp              132
sat-expak         132
wsn               132
visa      

In [10]:
data.service.value_counts()

-           141321
dns          68661
http         27011
smtp          6909
ftp-data      5391
ftp           4980
pop3          1528
ssh           1506
dhcp           120
snmp           109
ssl             86
irc             30
radius          21
Name: service, dtype: int64

services are not available for records 141321
Assume it as missing value and replace it with None

In [11]:
data.service= np.where(data.service == '-', 'None', data.service)

In [12]:
data.service.value_counts()

None        141321
dns          68661
http         27011
smtp          6909
ftp-data      5391
ftp           4980
pop3          1528
ssh           1506
dhcp           120
snmp           109
ssl             86
irc             30
radius          21
Name: service, dtype: int64

In [13]:
data.state.value_counts()

FIN    117164
INT    116438
CON     20134
REQ      3833
RST        84
ECO        12
ACC         4
URN         1
PAR         1
CLO         1
no          1
Name: state, dtype: int64

Assume that there could be chances of having the missing values in other features.
Let's automate the process to clean for other features

In [14]:
def Remove_dump_values(data, cols):
    for col in cols:
        data[col] = np.where(data[col].astype(str) == '-', 'None', data[col])
    return data

In [15]:
cols = data.columns
data_bin = Remove_dump_values(data, cols)

In [16]:
data_bin.head(10)

id                     dur proto service state spkts dpkts sbytes dbytes  \
0   1                 1.1e-05   udp    None   INT     2     0    496      0   
1   2                   8e-06   udp    None   INT     2     0   1762      0   
2   3                   5e-06   udp    None   INT     2     0   1068      0   
3   4                   6e-06   udp    None   INT     2     0    900      0   
4   5                   1e-05   udp    None   INT     2     0   2126      0   
5   6                   3e-06   udp    None   INT     2     0    784      0   
6   7                   6e-06   udp    None   INT     2     0   1960      0   
7   8  2.8000000000000003e-05   udp    None   INT     2     0   1384      0   
8   9                     0.0   arp    None   INT     1     0     46      0   
9  10                     0.0   arp    None   INT     1     0     46      0   

          rate sttl dttl         sload dload sloss dloss  \
0   90909.0902  254    0   180363632.0   0.0     0     0   
1  125000.0003  254    0   881000000.0   0.0     0     0   
2  200000.0051  254    0   854400000.0   0.0     0     0   
3  166666.6608  254    0   600000000.0   0.0     0     0   
4  100000.0025  254    0   850400000.0   0.0     0     0   
5  333333.3215  254    0  1045333312.0   0.0     0     0   
6  166666.6608  254    0  1306666624.0   0.0     0     0   
7  35714.28522  254    0   197714288.0   0.0     0     0   
8          0.0    0    0           0.0   0.0     0     0   
9          0.0    0    0           0.0   0.0     0     0   

                 sinpkt dinpkt sjit djit swin stcpb dtcpb dwin tcprtt synack  \
0  0.011000000000000001    0.0  0.0  0.0    0     0     0    0    0.0    0.0   
1                 0.008    0.0  0.0  0.0    0     0     0    0    0.0    0.0   
2                 0.005    0.0  0.0  0.0    0     0     0    0    0.0    0.0   
3                 0.006    0.0  0.0  0.0    0     0     0    0    0.0    0.0   
4                  0.01    0.0  0.0  0.0    0     0     0    0    0.0    0.0   
5                 0.003    0.0  0.0  0.0    0     0     0    0    0.0    0.0   
6                 0.006    0.0  0.0  0.0    0     0     0    0    0.0    0.0   
7  0.027999999999999997    0.0  0.0  0.0    0     0     0    0    0.0    0.0   
8    60000.687999999995    0.0  0.0  0.0    0     0     0    0    0.0    0.0   
9             60000.712    0.0  0.0  0.0    0     0     0    0    0.0    0.0   

  ackdat smean dmean trans_depth response_body_len ct_srv_src ct_state_ttl  \
0    0.0   248     0           0                 0          2            2   
1    0.0   881     0           0                 0          2            2   
2    0.0   534     0           0                 0          3            2   
3    0.0   450     0           0                 0          3            2   
4    0.0  1063     0           0                 0          3            2   
5    0.0   392     0           0                 0          2            2   
6    0.0   980     0           0                 0          2            2   
7    0.0   692     0           0                 0          3            2   
8    0.0    46     0           0                 0          2            2   
9    0.0    46     0           0                 0          2            2   

  ct_dst_ltm ct_src_dport_ltm ct_dst_sport_ltm ct_dst_src_ltm is_ftp_login  \
0          1                1                1              2            0   
1          1                1                1              2            0   
2          1                1                1              3            0   
3          2                2                1              3            0   
4          2                2                1              3            0   
5          2                2                1              2            0   
6          2                2                1              2            0   
7          1                1                1              3            0   
8          2                2 

 #### Remove unnecessary features

 Remove the ID feature as it is unique for every record

In [17]:
data_bin = data_bin.drop(['id'], axis=1)

In [18]:
attack_cat = data_bin.loc[:,"attack_cat"]

In [19]:
len(attack_cat), len(data_bin)

(257673, 257673)

In [20]:
data_bin.drop(['attack_cat'], axis=1, inplace=True)

In [21]:
data_bin.columns

Index(['dur', 'proto', 'service', 'state', 'spkts', 'dpkts', 'sbytes',
       'dbytes', 'rate', 'sttl', 'dttl', 'sload', 'dload', 'sloss', 'dloss',
       'sinpkt', 'dinpkt', 'sjit', 'djit', 'swin', 'stcpb', 'dtcpb', 'dwin',
       'tcprtt', 'synack', 'ackdat', 'smean', 'dmean', 'trans_depth',
       'response_body_len', 'ct_srv_src', 'ct_state_ttl', 'ct_dst_ltm',
       'ct_src_dport_ltm', 'ct_dst_sport_ltm', 'ct_dst_src_ltm',
       'is_ftp_login', 'ct_ftp_cmd', 'ct_flw_http_mthd', 'ct_src_ltm',
       'ct_srv_dst', 'is_sm_ips_ports', 'label'],
      dtype='object')

In [22]:
data_bin.dtypes

dur                  object
proto                object
service              object
state                object
spkts                object
dpkts                object
sbytes               object
dbytes               object
rate                 object
sttl                 object
dttl                 object
sload                object
dload                object
sloss                object
dloss                object
sinpkt               object
dinpkt               object
sjit                 object
djit                 object
swin                 object
stcpb                object
dtcpb                object
dwin                 object
tcprtt               object
synack               object
ackdat               object
smean                object
dmean                object
trans_depth          object
response_body_len    object
ct_srv_src           object
ct_state_ttl         object
ct_dst_ltm           object
ct_src_dport_ltm     object
ct_dst_sport_ltm     object
ct_dst_src_ltm      

In [23]:
from sklearn.preprocessing import LabelEncoder 
df_ = data_bin.copy()
print(df_.dtypes)

le = LabelEncoder() 
df_first = df_.copy()  
df_['proto']= le.fit_transform(df_['proto']) 
df_['service']= le.fit_transform(df_['service'])
df_['state']= le.fit_transform(df_['state']) 

dur                  object
proto                object
service              object
state                object
spkts                object
dpkts                object
sbytes               object
dbytes               object
rate                 object
sttl                 object
dttl                 object
sload                object
dload                object
sloss                object
dloss                object
sinpkt               object
dinpkt               object
sjit                 object
djit                 object
swin                 object
stcpb                object
dtcpb                object
dwin                 object
tcprtt               object
synack               object
ackdat               object
smean                object
dmean                object
trans_depth          object
response_body_len    object
ct_srv_src           object
ct_state_ttl         object
ct_dst_ltm           object
ct_src_dport_ltm     object
ct_dst_sport_ltm     object
ct_dst_src_ltm      

In [24]:
df_.head()

dur  proto  service  state spkts dpkts sbytes dbytes         rate sttl  \
0  1.1e-05    119        0      5     2     0    496      0   90909.0902  254   
1    8e-06    119        0      5     2     0   1762      0  125000.0003  254   
2    5e-06    119        0      5     2     0   1068      0  200000.0051  254   
3    6e-06    119        0      5     2     0    900      0  166666.6608  254   
4    1e-05    119        0      5     2     0   2126      0  100000.0025  254   

  dttl        sload dload sloss dloss                sinpkt dinpkt sjit djit  \
0    0  180363632.0   0.0     0     0  0.011000000000000001    0.0  0.0  0.0   
1    0  881000000.0   0.0     0     0                 0.008    0.0  0.0  0.0   
2    0  854400000.0   0.0     0     0                 0.005    0.0  0.0  0.0   
3    0  600000000.0   0.0     0     0                 0.006    0.0  0.0  0.0   
4    0  850400000.0   0.0     0     0                  0.01    0.0  0.0  0.0   

  swin stcpb dtcpb dwin tcprtt synack ackdat smean dmean trans_depth  \
0    0     0     0    0    0.0    0.0    0.0   248     0           0   
1    0     0     0    0    0.0    0.0    0.0   881     0           0   
2    0     0     0    0    0.0    0.0    0.0   534     0           0   
3    0     0     0    0    0.0    0.0    0.0   450     0           0   
4    0     0     0    0    0.0    0.0    0.0  1063     0           0   

  response_body_len ct_srv_src ct_state_ttl ct_dst_ltm ct_src_dport_ltm  \
0                 0          2            2          1                1   
1                 0          2            2          1                1   
2                 0          3            2          1                1   
3                 0          3            2          2                2   
4                 0          3            2          2                2   

  ct_dst_sport_ltm ct_dst_src_ltm is_ftp_login ct_ftp_cmd ct_flw_http_mthd  \
0                1              2            0          0                0   
1                1              2            0          0                0   
2                1              3            0          0                0   
3                1              3            0          0                0   
4                1              3            0          0                0   

  ct_src_ltm ct_srv_dst is_sm_ips_ports label  
0          1          2               0     0  
1          1          2               0     0  
2          1          3               0     0  
3          2          3               0     0  
4          2          3               0     0

In [25]:
df_ = df_.replace('nan',np.NaN)
df_.isna().sum()

dur                  0
proto                0
service              0
state                0
spkts                0
dpkts                0
sbytes               0
dbytes               0
rate                 0
sttl                 0
dttl                 0
sload                0
dload                0
sloss                0
dloss                0
sinpkt               0
dinpkt               0
sjit                 0
djit                 0
swin                 0
stcpb                0
dtcpb                0
dwin                 0
tcprtt               0
synack               0
ackdat               0
smean                0
dmean                0
trans_depth          0
response_body_len    0
ct_srv_src           0
ct_state_ttl         0
ct_dst_ltm           0
ct_src_dport_ltm     0
ct_dst_sport_ltm     0
ct_dst_src_ltm       0
is_ftp_login         0
ct_ftp_cmd           0
ct_flw_http_mthd     0
ct_src_ltm           0
ct_srv_dst           0
is_sm_ips_ports      0
label                0
dtype: int6

In [26]:
df_.dropna(inplace=True)

In [27]:
new_df =  df_.apply(pd.to_numeric)
print(len(new_df))
new_df.dtypes

257673


dur                  float64
proto                  int64
service                int64
state                  int64
spkts                  int64
dpkts                  int64
sbytes                 int64
dbytes                 int64
rate                 float64
sttl                   int64
dttl                   int64
sload                float64
dload                float64
sloss                  int64
dloss                  int64
sinpkt               float64
dinpkt               float64
sjit                 float64
djit                 float64
swin                   int64
stcpb                  int64
dtcpb                  int64
dwin                   int64
tcprtt               float64
synack               float64
ackdat               float64
smean                  int64
dmean                  int64
trans_depth            int64
response_body_len      int64
ct_srv_src             int64
ct_state_ttl           int64
ct_dst_ltm             int64
ct_src_dport_ltm       int64
ct_dst_sport_l

### Normalise all the values in the dataset

In [28]:
label = new_df.label
features = new_df.iloc[:,:-1]

new_df.shape, features.shape, label.shape

((257673, 43), (257673, 42), (257673,))

In [29]:
from sklearn.model_selection import train_test_split


features_train, features_test, labels_train, labels_test = train_test_split(features, label, stratify = label, \
                                                                            test_size=0.4, random_state=34)

In [30]:
from sklearn.pipeline import Pipeline
from sklearn.model_selection import GridSearchCV
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split, StratifiedShuffleSplit
from sklearn.svm import SVC

scaler = StandardScaler()

clf_svm = SVC(C=0.1, gamma='auto', max_iter=5000 )

# pipeline = Pipeline(steps = [("scaling", scaler), ("clf",clf_svm )])
# param_grid = {
#                 "clf__C":[0.1,1,10],
#               "clf__max_iter": [1000]
#              }

#sss = StratifiedShuffleSplit(n_splits=3, test_size=0.20, random_state=23)

#grid = GridSearchCV(pipeline, param_grid, verbose = 1, cv = sss, scoring = 'f1', return_train_score=True)
#grid.fit(features_train, labels_train)

clf_svm.fit(features_train, labels_train)

predictions_svm = clf_svm.predict(features_test)

In [31]:
grid.best_estimator_

NameError: name 'grid' is not defined

In [ ]:
from sklearn.metrics import confusion_matrix, classification_report

print ("\naccuracy of svm with grid search: {:.2f}\n".format((clf_svm.score(features_test, labels_test)*100)) )
print (classification_report(labels_test, predictions_svm))
print (confusion_matrix(labels_test, predictions_svm))

In [ ]:
### Convert SKlearn model to ADSModel

In [ ]:
from ads.common.model import ADSModel
svm_model = ADSModel.from_estimator(clf_svm)

In [ ]:
from ads.catalog.model import ModelSummaryList, ModelCatalog
from ads.catalog.project import ProjectSummaryList, ProjectCatalog
from ads.catalog.summary import SummaryList
from ads.common.model_artifact import ModelArtifact

In [ ]:
path_to_model_artifact = "/home/datascience/new_fn"
model_artifact = svm_model.prepare(path_to_model_artifact, fn_artifact_files_included= True, force_overwrite=True)


In [ ]:
import os

print(os.environ["NB_SESSION_COMPARTMENT_OCID"])
print(os.environ["PROJECT_OCID"])
print(os.environ["USER_OCID"])
print(os.environ["TENANCY_OCID"])
print(os.environ["NB_REGION"])

In [ ]:
compartment_id = os.environ["NB_SESSION_COMPARTMENT_OCID"]
project_id = os.environ["PROJECT_OCID"]


In [ ]:
model = model_artifact.save(project_id = project_id, compartment_id=compartment_id, display_name="new_fn",
                                     description="predicting attack", training_script_path = "training.ipynb", ignore_pending_changes=True)
